src/TreeTest.cpp

```diff
@@ -66,3 +68,33 @@ TEST(Tree, GetPath) {
 	tree->getPath(path, tree->lastNode(), false);
 	ASSERT_TRUE(path.size() > 1);
 }
+
+TEST(Tree, ASC) {
+	//test adaptive stepsize control
+	Tree<Vector2f> *tree = TreeFor2dPlane(
+		make_shared<GridStateSpace>(50, 50, 50, 50),
+		Vector2f(40, 40),	//	goal point
+		5);					//	step size
+
+	//	give it plenty of iterations so it's not likely to fail
+	const int maxIterations = 10000;
+	tree->setMaxIterations(maxIterations);
+	tree->setGoalMaxDist(5);
+	tree->setASCEnabled(true);
+
+	tree->setStartState(Vector2f(10, 10));
+	bool success = tree->run();	//	run with the given starting point
+	ASSERT_TRUE(success);
+
+	vector<Vector2f> path;
+	tree->getPath(path, tree->lastNode(), true);
+
+	bool varied = false;
+	for (int i = 1; !varied && i < path.size() - 2; i++) {
+		Vector2f x = path[i] - path[i - 1];
+		Vector2f y = path[i] - path[i + 1];
+		float n = MAGNITUDE(x) / MAGNITUDE(y);
```

Comment:
Eigen's Vector2F class has a `norm()` method that does the same thing as `MAGNITUDE()`